services/gitlab/gitlab-merge-requests.service.js

```diff
@@ -0,0 +1,350 @@
+import Joi from 'joi'
+import { optionalUrl, nonNegativeInteger } from '../validators.js'
+import { metric } from '../text-formatters.js'
+import GitLabBase from './gitlab-base.js'
+
+// The total number of MR is in the `x-total` field in the headers.
+// https://docs.gitlab.com/ee/api/index.html#other-pagination-headers
+const schema = Joi.object({
+  'x-total': Joi.number().integer(),
+  'x-page': nonNegativeInteger,
+})
+
+const queryParamSchema = Joi.object({
+  labels: Joi.string(),
+  gitlab_url: optionalUrl,
+}).required()
+
+const documentation = `
+<p>
+  You may use your GitLab Project Id (e.g. 278964) or your Project Path (e.g. gitlab-org/gitlab ).
+  Note that only internet-accessible GitLab instances are supported, for example https://jihulab.com, https://gitlab.gnome.org, or https://gitlab.com/.
+</p>
```

Comment:
Just a note for a potential follow up PR, but we may want to encapsulate this doc text in one place and then reuse it, similar to what we've done with the GitHub class tree (it seems like we've got duplicate copies of this on multiple GitLab classes nowadays)

Reply:
TODO, note at https://github.com/badges/shields/issues/8077#issuecomment-1207597218